src/response.rs

```diff
@@ -580,13 +587,13 @@ fn read_next_line<R: Read>(reader: &mut R) -> IoResult<String> {
 }
 
 /// Limits a `Read` to a content size (as set by a "Content-Length" header).
-struct LimitedRead<R> {
+struct LimitedRead<R: Read> {
```

Comment:
Good point. Mainly I had found it confusing that you could create a LimitedRead that wraps something `!Read`, but then it just wouldn't do anything. But we can achieve that constraint by limiting `new`, since the fields of this struct are private. I'm guessing a type constraint on the struct itself would be important for structs that have public fields and can be constructed without going through `new`.